types/context_cache.go

```diff
@@ -1,33 +1,38 @@
 package types
 
 import (
+	"strings"
 	"sync"
 
+	"github.com/armon/go-metrics"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
 const (
-	MESSAGE_COUNT	= "message_count"
-	TX_COUNT		= "transaction_count"
-	ORDER_COUNT		= "order_count"
+	MESSAGE_COUNT      = "message_count"
```

Comment:
this implementation is fine, but another idea is to add the type label to this metric, thoughts?